From 264822f3a8a7be49cf643b60289d86e0d61fd150 Mon Sep 17 00:00:00 2001
From: David Milburn <dmilburn@redhat.com>
Date: Tue, 15 Jul 2014 20:37:21 -0400
Subject: [mmc] sdhci-pci: break out definitions to header file

Message-id: <1405456643-17890-2-git-send-email-dmilburn@redhat.com>
Patchwork-id: 85840
O-Subject: [RHEL7.1 PATCH BZ 968020 1/3] mmc: sdhci-pci: break out definitions to header file
Bugzilla: 968020
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

commit 522624f97ee22684cf1b169b5a490cc3ad87b22c
Author: Adam Lee <adam.lee@canonical.com>
Date:   Wed Dec 18 22:23:38 2013 +0800

    mmc: sdhci-pci: break out definitions to header file

    Break out definitions in sdhci-pci.c to sdhci-pci.h, for introducing
    module files like sdhci-pci-xxx.c

    Signed-off-by: Adam Lee <adam.lee@canonical.com>
    Signed-off-by: Chris Ball <chris@printf.net>

RHEL: This patch only moves the device IDs currently supported by RHEL
and the RHEL struct sdhci_pci_slot does not have hw_reset.

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/mmc/host/sdhci-pci.c b/drivers/mmc/host/sdhci-pci.c
index 701d06d..649542b 100644
--- a/drivers/mmc/host/sdhci-pci.c
+++ b/drivers/mmc/host/sdhci-pci.c
@@ -27,70 +27,7 @@
 #include <linux/mmc/sdhci-pci-data.h>
 
 #include "sdhci.h"
-
-/*
- * PCI device IDs
- */
-#define PCI_DEVICE_ID_INTEL_PCH_SDIO0 0x8809
-#define PCI_DEVICE_ID_INTEL_PCH_SDIO1 0x880a
-#define PCI_DEVICE_ID_INTEL_BYT_EMMC 0x0f14
-#define PCI_DEVICE_ID_INTEL_BYT_SDIO 0x0f15
-#define PCI_DEVICE_ID_INTEL_BYT_SD 0x0f16
-
-/*
- * PCI registers
- */
-
-#define PCI_SDHCI_IFPIO   0x00
-#define PCI_SDHCI_IFDMA   0x01
-#define PCI_SDHCI_IFVENDOR  0x02
-
-#define PCI_SLOT_INFO   0x40 /* 8 bits */
-#define  PCI_SLOT_INFO_SLOTS(x)  ((x >> 4) & 7)
-#define  PCI_SLOT_INFO_FIRST_BAR_MASK 0x07
-
-#define MAX_SLOTS   8
-
-struct sdhci_pci_chip;
-struct sdhci_pci_slot;
-
-struct sdhci_pci_fixes {
- unsigned int  quirks;
- unsigned int  quirks2;
- bool   allow_runtime_pm;
-
- int   (*probe) (struct sdhci_pci_chip *);
-
- int   (*probe_slot) (struct sdhci_pci_slot *);
- void   (*remove_slot) (struct sdhci_pci_slot *, int);
-
- int   (*suspend) (struct sdhci_pci_chip *);
- int   (*resume) (struct sdhci_pci_chip *);
-};
-
-struct sdhci_pci_slot {
- struct sdhci_pci_chip *chip;
- struct sdhci_host *host;
- struct sdhci_pci_data *data;
-
- int   pci_bar;
- int   rst_n_gpio;
- int   cd_gpio;
- int   cd_irq;
-};
-
-struct sdhci_pci_chip {
- struct pci_dev  *pdev;
-
- unsigned int  quirks;
- unsigned int  quirks2;
- bool   allow_runtime_pm;
- const struct sdhci_pci_fixes *fixes;
-
- int   num_slots; /* Slots on controller */
- struct sdhci_pci_slot *slots[MAX_SLOTS]; /* Pointers to host slots */
-};
-
+#include "sdhci-pci.h"
 
 /*****************************************************************************\
  *                                                                           *
diff --git a/drivers/mmc/host/sdhci-pci.h b/drivers/mmc/host/sdhci-pci.h
new file mode 100644
index 0000000..34081c0
--- /dev/null
+++ b/drivers/mmc/host/sdhci-pci.h
@@ -0,0 +1,70 @@
+#ifndef __SDHCI_PCI_H
+#define __SDHCI_PCI_H
+
+/*
+ * PCI device IDs
+ */
+
+#define PCI_DEVICE_ID_INTEL_PCH_SDIO0 0x8809
+#define PCI_DEVICE_ID_INTEL_PCH_SDIO1 0x880a
+#define PCI_DEVICE_ID_INTEL_BYT_EMMC 0x0f14
+#define PCI_DEVICE_ID_INTEL_BYT_SDIO 0x0f15
+#define PCI_DEVICE_ID_INTEL_BYT_SD 0x0f16
+
+/*
+ * PCI registers
+ */
+
+#define PCI_SDHCI_IFPIO   0x00
+#define PCI_SDHCI_IFDMA   0x01
+#define PCI_SDHCI_IFVENDOR  0x02
+
+#define PCI_SLOT_INFO   0x40 /* 8 bits */
+#define  PCI_SLOT_INFO_SLOTS(x)  ((x >> 4) & 7)
+#define  PCI_SLOT_INFO_FIRST_BAR_MASK 0x07
+
+#define MAX_SLOTS   8
+
+struct sdhci_pci_chip;
+struct sdhci_pci_slot;
+
+struct sdhci_pci_fixes {
+ unsigned int  quirks;
+ unsigned int  quirks2;
+ bool   allow_runtime_pm;
+
+ int   (*probe) (struct sdhci_pci_chip *);
+
+ int   (*probe_slot) (struct sdhci_pci_slot *);
+ void   (*remove_slot) (struct sdhci_pci_slot *, int);
+
+ int   (*suspend) (struct sdhci_pci_chip *);
+ int   (*resume) (struct sdhci_pci_chip *);
+};
+
+struct sdhci_pci_slot {
+ struct sdhci_pci_chip *chip;
+ struct sdhci_host *host;
+ struct sdhci_pci_data *data;
+
+ int   pci_bar;
+ int   rst_n_gpio;
+ int   cd_gpio;
+ int   cd_irq;
+
+ void (*hw_reset)(struct sdhci_host *host);
+};
+
+struct sdhci_pci_chip {
+ struct pci_dev  *pdev;
+
+ unsigned int  quirks;
+ unsigned int  quirks2;
+ bool   allow_runtime_pm;
+ const struct sdhci_pci_fixes *fixes;
+
+ int   num_slots; /* Slots on controller */
+ struct sdhci_pci_slot *slots[MAX_SLOTS]; /* Pointers to host slots */
+};
+
+#endif /* __SDHCI_PCI_H */
-- 
1.7.1